integration_test/ops_agent_test.go

```diff
@@ -2091,6 +2093,386 @@ func TestPrometheusMetricsWithJSONExporter(t *testing.T) {
+			// script should finish without error
+			if err != nil {
+				t.Fatalf("failed to run the http server in VM with err: %v, stderr: %s", err, setupOut.Stderr)
+			}
+			// Wait until the http server is ready
+			time.Sleep(30 * time.Second)
+			liveCheckOut, liveCheckErr := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", `curl "http://localhost:8000/data.json"`)
+			// We will retry when the Python HTTP server is not started:
+			// the stderr will have: "curl: (7) Failed to connect to localhost port 8000 after 1 ms: Connection refused"
+			// Otherwise - break the retrying loop
+			if liveCheckErr == nil && !strings.Contains(liveCheckOut.Stderr, "Connection refused") {
+				break
+			}
+
+			// Out of attempts
+			if attempt == maxAttampts-1 {
+				errString := fmt.Sprintf("last stdout: %s last stderr: %s", liveCheckOut.Stdout, liveCheckOut.Stderr)
+				if liveCheckErr != nil {
+					errString = fmt.Sprintf("last err: %v %s", liveCheckErr, errString)
+				}
+				t.Fatalf("failed to start the HTTP server - exhausted retries. %s", errString)
+			}
+		}
+
+		// 3. Config and start the agent
+		// Set the scrape interval to 10 second, so that metrics points can be
+		// received faster to shorten the duration of this test
+		config := `metrics:
+  receivers:
+    prom_app:
+      type: prometheus
+      config:
+        scrape_configs:
+        - job_name: json
+          metrics_path: /data.json
+          scrape_interval: 10s
+          static_configs:
+            - targets:
+              - localhost:8000
+  service:
+    pipelines:
+      prom_pipeline:
+        receivers: [prom_app]
+`
+		// Turn on the prometheus feature gate.
+		if err := gce.SetEnvironmentVariables(ctx, logger.ToMainLog(), vm, map[string]string{"UNSUPPORTED_BETA_PROMETHEUS_RECEIVER": "enabled"}); err != nil {
+			t.Fatal(err)
+		}
+		if err := setupOpsAgent(ctx, logger, vm, config); err != nil {
+			t.Fatal(err)
+		}
+
+		// Wait long enough for the data to percolate through the backends
+		// under normal circumstances. Based on some experiments, 2 minutes
+		// is normal; wait a bit longer to be on the safe side.
+		time.Sleep(3 * time.Minute)
+		window := time.Minute
+		var multiErr error
+
+		// 4. Wait for the initial set of metrics and check
+		// For Histogram: We use prometheus.LinearBuckets(0, 20, 5), to have
+		// buckets w/ le=[0, 20, 40, 60, 80] plus the +inf final bucket
+		// For step 1, we observe points [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
+		// And get:
+		// Bounds (less than or equal) |0  |20     |40     |60     |80     |+inf
+		// Points                      |[0]|[10,20]|[30,40]|[50,60]|[70,80]|[90]
+		// Count                       |1  |2      |2      |2      |2      |1
+		// And histogram metrics are stored as cumulative type metrics, so for
+		// this initial step, Count/Mean/SumOfSquaredDeviation are all zeros,
+		// and the BucketCounts is nil
+		stepOneExpectedHistogram := &distribution.Distribution{
```

Comment:
I don't completely understand why these are all 0 values? They all do have some values right (in your example above, they have a total of 10 points). Shouldn't this have the first values? I see in step 2 they get the delta between the second and the first set of values, but in this case the delta between the first and 0 is dropped?

Reply:
If this is expected, please let me know why. I may be missing something

Reply:
Sync'ed offline, the cumulative points (histogram, count & sum of summary) get normalized when exporting to GCM: the first point of a cumulative time series is not sent and used to subtract from all following points. That's why even though the `first step` point has values, the agent scrape this multiple times and they are all received as 0 values. Added comments to point to the normalization of the cumulative points.

Reply:
This makes sense to me. I think we should atleast manually verify that this is actually what promql users expect once they migrate off of prometheus. We should be doing what GMP does on GKE, or what prometheus does by default